spinalcordtoolbox/deepseg/models.py

```diff
@@ -95,23 +95,6 @@
         "contrasts": ["mp2rage"],
         "default": False,
     },
-    "model_seg_ms_lesion_mp2rage": {
-        "url": {
-            "seed1": ["https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20230210/model_seg_lesion_mp2rage_r20230210_dil32_seed01.zip"],
-            "seed2": ["https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20230210/model_seg_lesion_mp2rage_r20230210_dil32_seed02.zip"],
-            "seed3": ["https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20230210/model_seg_lesion_mp2rage_r20230210_dil32_seed03.zip"],
-            "seed4": ["https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20230210/model_seg_lesion_mp2rage_r20230210_dil32_seed04.zip"],
-            "seed5": ["https://github.com/ivadomed/model_seg_ms_mp2rage/releases/download/r20230210/model_seg_lesion_mp2rage_r20230210_dil32_seed05.zip"],
-        },
```

Comment:
Note: Since the old model was the only one to use "ensemble" inference, and because "ensemble" inference was only ever implemented for the `ivadomed` models, then that means that our ensemble code is now "dead".

CC: @jcohenadad, @mguaypaq How likely is it that we create model ensembles in the future, and is the ensemble logic worth maintaining?

Personally, I think because we could easily implement ensembling in the future for non-ivadomed models, I think it might be worth leaving the ensembling code as-is for now? (Since, most of the ensembling code is "general", and it would be a bit dangerous IMO to remove the ensembling logic only to have to re-add it later on, since it involves finicky tweaks in quite a few places.)

Reply:
For posterity, here is a quick list of places where ensembling is considered:

- "General" code
  - `models.py :: find_model_folder_paths`
  - `models.py :: is_up_to_date`
  - `models.py :: is_valid`
  - `models.py :: install_model`
- ivadomed-specific code:
  - `inference.py :: segment_and_average_volumes`

Reply:
I think the ensemble model logic is worth keeping around.

I typically like to remove dead code, because:
1. it's untested, and
2. it puts constraints on the rest of the code, which has to stay compatible.

Here point 1 still applies, but we know that it was successfully used at least once, so that mitigates it.

For point 2, extra constraints on the rest of the code are usually annoying, but here I think they're beneficial: the dead code actually serves as a precise, executable form of documentation for the kinds of things that we need to keep in mind in the future, if we want to stay compatible with ensemble models with our data formats, file and folder naming, etc.